### ISSUE TITLE
while running plan command for FileShare and Blob getting error

### ISSUE THREAD

**Issue by dshah93** [hashicorp/terraform-provider-azurerm#25485]
### Is there an existing issue for this?

- [x] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

### Terraform Version

1.7.5

### AzureRM Provider Version

3.97.1

### Affected Resource(s)/Data Source(s)

azurerm_private_endpoint.private_endpoint and azurerm_storage_share.azure_storage_share

### Terraform Configuration Files

```hcl
trying to create a Private Endpoint for Storage File/Blob/Table/Queue in for loop structure.
trying to create a file share in for loop structure.
```

### Debug Output/Panic Output

```shell
2024-04-02T10:28:38.7847015Z [31m╷[0m[0m
2024-04-02T10:28:38.7848412Z [31m│[0m [0m[1m[31mError: [0m[0m[1mPlugin did not respond[0m
2024-04-02T10:28:38.7849097Z [31m│[0m [0m
2024-04-02T10:28:38.7849889Z [31m│[0m [0m[0m  with module.azure_storage_share["fx-csc-dna-use-p-ase-test"].azurerm_storage_share.azure_storage_share,
2024-04-02T10:28:38.7850847Z [31m│[0m [0m  on ..\..\..\modules\azure_storage_share\main.tf line 1, in resource "azurerm_storage_share" "azure_storage_share":
2024-04-02T10:28:38.7851669Z [31m│[0m [0m   1: resource "azurerm_storage_share" "azure_storage_share" [4m{[0m[0m
2024-04-02T10:28:38.7852197Z [31m│[0m [0m
2024-04-02T10:28:38.7854990Z [31m│[0m [0mThe plugin encountered an error, and failed to respond to the
2024-04-02T10:28:38.7855775Z [31m│[0m [0mplugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may
2024-04-02T10:28:38.7856424Z [31m│[0m [0mcontain more details.
2024-04-02T10:28:38.7856915Z [31m╵[0m[0m
2024-04-02T10:28:38.7857460Z [31m╷[0m[0m
2024-04-02T10:28:38.7858040Z [31m│[0m [0m[1m[31mError: [0m[0m[1mPlugin did not respond[0m
2024-04-02T10:28:38.7858592Z [31m│[0m [0m
2024-04-02T10:28:38.7859345Z [31m│[0m [0m[0m  with module.file_private_endpoint["sacscdnauseprdintlxfx00"].azurerm_private_endpoint.private_endpoint,
2024-04-02T10:28:38.7861967Z [31m│[0m [0m  on ..\..\..\modules\private_link_service\modules\private_endpoint\main.tf line 1, in resource "azurerm_private_endpoint" "private_endpoint":
2024-04-02T10:28:38.7864809Z [31m│[0m [0m   1: resource "azurerm_private_endpoint" "private_endpoint" [4m{[0m[0m
2024-04-02T10:28:38.7865483Z [31m│[0m [0m
2024-04-02T10:28:38.7867097Z [31m│[0m [0mThe plugin encountered an error, and failed to respond to the
2024-04-02T10:28:38.7867623Z [31m│[0m [0mplugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may
2024-04-02T10:28:38.7868180Z [31m│[0m [0mcontain more details.
2024-04-02T10:28:38.7868623Z [31m╵[0m[0m
2024-04-02T10:28:38.7869105Z [31m╷[0m[0m
2024-04-02T10:28:38.7869614Z [31m│[0m [0m[1m[31mError: [0m[0m[1mPlugin did not respond[0m
2024-04-02T10:28:38.7870414Z [31m│[0m [0m
2024-04-02T10:28:38.7871088Z [31m│[0m [0m[0m  with module.blob_private_endpoint["sacscdnauseprdintlxfx00"].azurerm_private_endpoint.private_endpoint,
2024-04-02T10:28:38.7872020Z [31m│[0m [0m  on ..\..\..\modules\private_link_service\modules\private_endpoint\main.tf line 1, in resource "azurerm_private_endpoint" "private_endpoint":
2024-04-02T10:28:38.7872888Z [31m│[0m [0m   1: resource "azurerm_private_endpoint" "private_endpoint" [4m{[0m[0m
2024-04-02T10:28:38.7873450Z [31m│[0m [0m
2024-04-02T10:28:38.7874014Z [31m│[0m [0mThe plugin encountered an error, and failed to respond to the
2024-04-02T10:28:38.7874703Z [31m│[0m [0mplugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may
2024-04-02T10:28:38.7875110Z [31m│[0m [0mcontain more details.
2024-04-02T10:28:38.7875398Z [31m╵[0m[0m
2024-04-02T10:28:38.8058213Z 
2024-04-02T10:28:38.8059320Z Stack trace from the terraform-provider-azurerm_v3.97.1_x5.exe plugin:
2024-04-02T10:28:38.8059785Z 
2024-04-02T10:28:38.8060237Z panic: runtime error: invalid memory address or nil pointer dereference
2024-04-02T10:28:38.8060989Z [signal 0xc0000005 code=0x0 addr=0x38 pc=0x43d3abf]
2024-04-02T10:28:38.8061270Z 
2024-04-02T10:28:38.8061607Z goroutine 63 [running]:
2024-04-02T10:28:38.8062347Z github.com/tombuildsstuff/giovanni/storage/2023-11-03/file/shares.Client.GetProperties({0xc00179ef40?}, {0x8cf6058, 0xc0014f9960}, {0xc004ae8200, 0x19})
2024-04-02T10:28:38.8063185Z 	github.com/tombuildsstuff/giovanni@v0.25.3/storage/2023-11-03/file/shares/properties_get.go:56 +0x31f
2024-04-02T10:28:38.8064088Z github.com/hashicorp/terraform-provider-azurerm/internal/services/storage/shim.DataPlaneStorageShareWrapper.Exists({0xc0012d2630?}, {0x8cf6058?, 0xc0014f9960?}, {0xc004ae8200?, 0xc000cd3608?})
2024-04-02T10:28:38.8065075Z 	github.com/hashicorp/terraform-provider-azurerm/internal/services/storage/shim/shares_data_plane.go:41 +0x3a
2024-04-02T10:28:38.8065906Z github.com/hashicorp/terraform-provider-azurerm/internal/services/storage.resourceStorageShareCreate(0x0?, {0x74bffe0?, 0xc001eea000?})
2024-04-02T10:28:38.8066665Z 	github.com/hashicorp/terraform-provider-azurerm/internal/services/storage/storage_share_resource.go:200 +0xc31
2024-04-02T10:28:38.8067528Z github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0x8cf5fb0?, {0x8cf5fb0?, 0xc0048fa510?}, 0xd?, {0x74bffe0?, 0xc001eea000?})
2024-04-02T10:28:38.8068323Z 	github.com/hashicorp/terraform-plugin-sdk/v2@v2.29.0/helper/schema/resource.go:766 +0x163
2024-04-02T10:28:38.8072508Z github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc001af5960, {0x8cf5fb0, 0xc0048fa510}, 0xc0014f1450, 0xc004ae6e80, {0x74bffe0, 0xc001eea000})
2024-04-02T10:28:38.8073296Z 	github.com/hashicorp/terraform-plugin-sdk/v2@v2.29.0/helper/schema/resource.go:909 +0xa89
2024-04-02T10:28:38.8075278Z github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc000297128, {0x8cf5fb0?, 0xc0048fa420?}, 0xc0005634f0)
2024-04-02T10:28:38.8094235Z 	github.com/hashicorp/terraform-plugin-sdk/v2@v2.29.0/helper/schema/grpc_provider.go:1060 +0xdbc
2024-04-02T10:28:38.8099211Z github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0xc00115cd20, {0x8cf5fb0?, 0xc004acba10?}, 0xc0014f8fc0)
2024-04-02T10:28:38.8100108Z 	github.com/hashicorp/terraform-plugin-go@v0.19.0/tfprotov5/tf5server/server.go:859 +0x56b
2024-04-02T10:28:38.8100791Z github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0x80e95c0?, 0xc00115cd20}, {0x8cf5fb0, 0xc004acba10}, 0xc0014f8f50, 0x0)
2024-04-02T10:28:38.8101450Z 	github.com/hashicorp/terraform-plugin-go@v0.19.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:467 +0x169
2024-04-02T10:28:38.8101952Z google.golang.org/grpc.(*Server).processUnaryRPC(0xc000468000, {0x8d1db00, 0xc0014ce340}, 0xc004aeaea0, 0xc001c250e0, 0xe5f1d38, 0x0)
2024-04-02T10:28:38.8102551Z 	google.golang.org/grpc@v1.58.3/server.go:1374 +0xde7
2024-04-02T10:28:38.8102925Z google.golang.org/grpc.(*Server).handleStream(0xc000468000, {0x8d1db00, 0xc0014ce340}, 0xc004aeaea0, 0x0)
2024-04-02T10:28:38.8103330Z 	google.golang.org/grpc@v1.58.3/server.go:1751 +0x9e7
2024-04-02T10:28:38.8103748Z google.golang.org/grpc.(*Server).serveStreams.func1.1()
2024-04-02T10:28:38.8104253Z 	google.golang.org/grpc@v1.58.3/server.go:986 +0xbb
2024-04-02T10:28:38.8104763Z created by google.golang.org/grpc.(*Server).serveStreams.func1 in goroutine 22
2024-04-02T10:28:38.8105253Z 	google.golang.org/grpc@v1.58.3/server.go:997 +0x145
2024-04-02T10:28:38.8105572Z 
2024-04-02T10:28:38.8106519Z Error: The terraform-provider-azurerm_v3.97.1_x5.exe plugin crashed!
2024-04-02T10:28:38.8106919Z 
2024-04-02T10:28:38.8108154Z This is always indicative of a bug within the plugin. It would be immensely
2024-04-02T10:28:38.8109022Z helpful if you could report the crash with the plugin's maintainers so that it
2024-04-02T10:28:38.8109543Z can be fixed. The output above should help diagnose the issue.
2024-04-02T10:28:38.8109848Z 
2024-04-02T10:28:38.8340582Z ##[error]Error: The process 'C:\a\_tool\terraform\1.7.5\x64\terraform.exe' failed with exit code 1
2024-04-02T10:28:38.8462282Z ##[section]Finishing: Terraform Apply
```

**Comment by dshah93** [comment]
While running plan command getting below error


2024-04-02T13:10:02.4280260Z [31m│[0m [0m[1m[31mError: [0m[0m[1mPlugin did not respond[0m
2024-04-02T13:10:02.4280639Z [31m│[0m [0m
2024-04-02T13:10:02.4282353Z [31m│[0m [0m[0m  with module.azure_storage_share["fx-csc-dna-use-p-ase-test"].azurerm_storage_share.azure_storage_share,
2024-04-02T13:10:02.4283535Z [31m│[0m [0m  on ..\..\..\modules\azure_storage_share\main.tf line 1, in resource "azurerm_storage_share" "azure_storage_share":
2024-04-02T13:10:02.4284389Z [31m│[0m [0m   1: resource "azurerm_storage_share" "azure_storage_share" [4m{[0m[0m
2024-04-02T13:10:02.4284956Z [31m│[0m [0m
2024-04-02T13:10:02.4285833Z [31m│[0m [0mThe plugin encountered an error, and failed to respond to the
2024-04-02T13:10:02.4286476Z [31m│[0m [0mplugin.(*GRPCProvider).ReadResource call. The plugin logs may contain more
2024-04-02T13:10:02.4287159Z [31m│[0m [0mdetails.
2024-04-02T13:10:02.4287537Z [31m╵[0m[0m
2024-04-02T13:10:02.4287848Z [31m╷[0m[0m
2024-04-02T13:10:02.4288299Z [31m│[0m [0m[1m[31mError: [0m[0m[1mPlugin did not respond[0m
2024-04-02T13:10:02.4288708Z [31m│[0m [0m
2024-04-02T13:10:02.4289359Z [31m│[0m [0m[0m  with module.storage_function_app_ase["sacscdnauseprdintlxfx00"].azurerm_storage_account.blob,
2024-04-02T13:10:02.4290281Z [31m│[0m [0m  on ..\..\..\modules\blob_storage\main.tf line 1, in resource "azurerm_storage_account" "blob":
2024-04-02T13:10:02.4291004Z [31m│[0m [0m   1: resource "azurerm_storage_account" "blob" [4m{[0m[0m
2024-04-02T13:10:02.4291373Z [31m│[0m [0m
2024-04-02T13:10:02.4292076Z [31m│[0m [0mThe plugin encountered an error, and failed to respond to the
2024-04-02T13:10:02.4292965Z [31m│[0m [0mplugin.(*GRPCProvider).ReadResource call. The plugin logs may contain more
2024-04-02T13:10:02.4293503Z [31m│[0m [0mdetails.
2024-04-02T13:10:02.4293914Z [31m╵[0m[0m
2024-04-02T13:10:02.4527863Z 
2024-04-02T13:10:02.4529910Z Stack trace from the terraform-provider-azurerm_v3.97.1_x5.exe plugin:
2024-04-02T13:10:02.4530777Z 
2024-04-02T13:10:02.4531966Z panic: runtime error: invalid memory address or nil pointer dereference
2024-04-02T13:10:02.4532682Z [signal 0xc0000005 code=0x0 addr=0x38 pc=0x47f3abf]
2024-04-02T13:10:02.4533103Z 
2024-04-02T13:10:02.4533579Z goroutine 1416 [running]:
2024-04-02T13:10:02.4535347Z github.com/tombuildsstuff/giovanni/storage/2023-11-03/file/shares.Client.GetProperties({0xc002553e20?}, {0x9116058, 0xc001711030}, {0xc0011ff6b5, 0x19})
2024-04-02T13:10:02.4536518Z 	github.com/tombuildsstuff/giovanni@v0.25.3/storage/2023-11-03/file/shares/properties_get.go:56 +0x31f
2024-04-02T13:10:02.4537843Z github.com/hashicorp/terraform-provider-azurerm/internal/services/storage/shim.DataPlaneStorageShareWrapper.Get({0xc00121a630?}, {0x9116058, 0xc001711030}, {0xc0011ff6b5, 0x19})
2024-04-02T13:10:02.4539057Z 	github.com/hashicorp/terraform-provider-azurerm/internal/services/storage/shim/shares_data_plane.go:52 +0x4a
2024-04-02T13:10:02.4540257Z github.com/hashicorp/terraform-provider-azurerm/internal/services/storage.resourceStorageShareRead(0xc00094b000, {0x78dffe0?, 0xc001b23680})
2024-04-02T13:10:02.4541429Z 	github.com/hashicorp/terraform-provider-azurerm/internal/services/storage/storage_share_resource.go:259 +0x431
2024-04-02T13:10:02.4542641Z github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).read(0x9115fb0?, {0x9115fb0?, 0xc002d939e0?}, 0xd?, {0x78dffe0?, 0xc001b23680?})
2024-04-02T13:10:02.4544460Z 	github.com/hashicorp/terraform-plugin-sdk/v2@v2.29.0/helper/schema/resource.go:783 +0x163
2024-04-02T13:10:02.4568742Z github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).RefreshWithoutUpgrade(0xc001acf500, {0x9115fb0, 0xc002d939e0}, 0xc001bb2ea0, {0x78dffe0, 0xc001b23680})
2024-04-02T13:10:02.4573015Z 	github.com/hashicorp/terraform-plugin-sdk/v2@v2.29.0/helper/schema/resource.go:1089 +0x552
2024-04-02T13:10:02.4574831Z github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ReadResource(0xc0006d18c0, {0x9115fb0?, 0xc002d938f0?}, 0xc00543e480)
2024-04-02T13:10:02.4576257Z 	github.com/hashicorp/terraform-plugin-sdk/v2@v2.29.0/helper/schema/grpc_provider.go:649 +0x48a
2024-04-02T13:10:02.4577842Z github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ReadResource(0xc0002d7040, {0x9115fb0?, 0xc002d93140?}, 0xc001e23680)
2024-04-02T13:10:02.4578841Z 	github.com/hashicorp/terraform-plugin-go@v0.19.0/tfprotov5/tf5server/server.go:789 +0x48b
2024-04-02T13:10:02.4580068Z github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ReadResource_Handler({0x85095c0?, 0xc0002d7040}, {0x9115fb0, 0xc002d93140}, 0xc001710930, 0x0)
2024-04-02T13:10:02.4581230Z 	github.com/hashicorp/terraform-plugin-go@v0.19.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:431 +0x169
2024-04-02T13:10:02.4582069Z google.golang.org/grpc.(*Server).processUnaryRPC(0xc0002ce1e0, {0x913db00, 0xc0014d6340}, 0xc0019a6fc0, 0xc001c0e4b0, 0xea11d08, 0x0)
2024-04-02T13:10:02.4582779Z 	google.golang.org/grpc@v1.58.3/server.go:1374 +0xde7
2024-04-02T13:10:02.4583424Z google.golang.org/grpc.(*Server).handleStream(0xc0002ce1e0, {0x913db00, 0xc0014d6340}, 0xc0019a6fc0, 0x0)
2024-04-02T13:10:02.4584106Z 	google.golang.org/grpc@v1.58.3/server.go:1751 +0x9e7
2024-04-02T13:10:02.4584688Z google.golang.org/grpc.(*Server).serveStreams.func1.1()
2024-04-02T13:10:02.4585253Z 	google.golang.org/grpc@v1.58.3/server.go:986 +0xbb
2024-04-02T13:10:02.4585873Z created by google.golang.org/grpc.(*Server).serveStreams.func1 in goroutine 27
2024-04-02T13:10:02.4586488Z 	google.golang.org/grpc@v1.58.3/server.go:997 +0x145
2024-04-02T13:10:02.4587025Z 
2024-04-02T13:10:02.4587789Z Error: The terraform-provider-azurerm_v3.97.1_x5.exe plugin crashed!
2024-04-02T13:10:02.4588307Z 
2024-04-02T13:10:02.4588843Z This is always indicative of a bug within the plugin. It would be immensely
2024-04-02T13:10:02.4589654Z helpful if you could report the crash with the plugin's maintainers so that it
2024-04-02T13:10:02.4590236Z can be fixed. The output above should help diagnose the issue.
2024-04-02T13:10:02.4590707Z 
2024-04-02T13:10:02.4908493Z ##[warning]Can't find loc string for key: TerraformPlanFailed
2024-04-02T13:10:02.4922045Z ##[error]Error: TerraformPlanFailed 1
2024-04-02T13:10:02.5020243Z ##[section]Finishing: Terraform Plan

**Comment by dshah93** [comment]
@rcskosir @manicminer can you look into this issue?

**Comment by tombuildsstuff** [comment]
hey @dshah93 

This issue is related to https://github.com/hashicorp/terraform-provider-azurerm/issues/25441 (which affected only Blobs) which was fixed by #25404 (which fixes this for all of the Storage Data Plane resources) - and as such a fix for this has been merged and will ship in this weeks release (and as such I've added this to this weeks milestone so a comment will be posted once the release becomes available).

In the interim you should be able to workaround this by pinning to the previous version of the AzureRM Provider (`v3.96.0`) - however since a fix for this one's been merged I'm going to close this issue for the moment.

Thanks!

**Comment by manicminer** [comment]
@dshah93 We found some more potential crashes that will be fixed by https://github.com/hashicorp/terraform-provider-azurerm/pull/25525, which is due to be released this week in v3.99.0

I believe this bug was actually introduced in v3.96.0, if you pin to v3.95.0 in the meantime this should avoid the crash you're experiencing. Thanks for your patience and your help on this!
